### PR TITLE
Support Camel Kafka Connect #254

### PR DESCRIPTION
**Pull request by apupier**
current limitation:
- properties file needs to be written with '=' without spaces
- reusing classic Camel catalog

Signed-off-by: Aurélien Pupier <apupier@redhat.com>

# Pull Request informations

## Rebase & Merge default requirements

1. Green build for master branch
2. Wait 24 hours after PR creation
3. Green job for PR
4. Approved PR

## PR labels default process

- READY_FOR_REVIEW  &rightarrow;  REVIEW_DONE  &rightarrow;  READY_FOR_MERGE

## Tests

- [ ] Are there **Unit tests**?
- [ ] Are there **Integration tests**?
- [ ] Do we need a new **UI test**?

## PR workflow progress

1. [ ] Tagged with relevant **PR labels**
2. [ ] Green **job for PR**
3. [ ] PR was created more than **24 hours ago** or **All committers approved** it
4. [ ] Green **master** branch build

### CODE REVIEW

**Comment by apupier** [comment]
also awaiting confirmation that I understood the functional need correctly, see https://github.com/camel-tooling/camel-language-server/issues/254#issuecomment-515914563

File: src/main/java/com/github/cameltooling/lsp/internal/instancemodel/PropertiesDSLModelHelper.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cameltooling.lsp.internal.instancemodel;
+
+public class PropertiesDSLModelHelper implements DSLModelHelper {
+	
+	String type;
```
Comment (lhein) [inline src/main/java/com/github/cameltooling/lsp/internal/instancemodel/PropertiesDSLModelHelper.java]: is there a specific reason to not make it private?

File: src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKafkaConnectDSLParser.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+public class CamelKafkaConnectDSLParser extends ParserFileHelper {
+	
+	public static final String CAMEL_SINK_URL = "camel.sink.url";
+	public static final String CAMEL_SOURCE_URL = "camel.source.url";
+	private static final Logger LOGGER = LoggerFactory.getLogger(CamelKafkaConnectDSLParser.class);
+
+	@Override
+	public String getCamelComponentUri(String line, int characterPosition) {
+		if (line.startsWith(CAMEL_SOURCE_URL) && CAMEL_SOURCE_URL.length() < characterPosition 
+				|| line.startsWith(CAMEL_SINK_URL) && CAMEL_SINK_URL.length() < characterPosition) {
```
Comment (lhein) [inline src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKafkaConnectDSLParser.java]: indentation problem?

Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKafkaConnectDSLParser.java]: seems to be the normal behavior with Eclipse defaults codestyle

I think it is displayed quite well with default size of tabs:
image

File: src/main/java/com/github/cameltooling/lsp/internal/instancemodel/PropertiesDSLModelHelper.java (hunk 1 above)
Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/instancemodel/PropertiesDSLModelHelper.java]: no, good catch

**Pull request merged**